softwareComponents/configuration/include/configuration/rofiworld.hpp

```diff
@@ -571,10 +571,42 @@ class RofiWorld {
     }
 
     /**
-     * \brief Get a container of ModuleInfo
+     * \brief Get a range of modules
      */
-    const auto& modules() const {
-        return _modules;
+    auto modules() const -> std::ranges::range auto
+    {
+        return std::views::transform( _modules,
+                                      []( const ModuleInfo & moduleInfo ) -> const Module & {
+                                          assert( moduleInfo.module );
+                                          return *moduleInfo.module;
+                                      } );
+    }
+
+    /**
+     * \brief Get a range of modules
+     */
+    auto modules() -> std::ranges::range auto
+    {
+        return std::views::transform( _modules,
+                                      []( const ModuleInfo & moduleInfo ) -> Module & {
+                                          assert( moduleInfo.module );
+                                          return *moduleInfo.module;
+                                      } );
+    }
+
+    /**
+     * \brief Get a range of modules with their absolute positions
+     *
+     * `this` has to be prepared.
+     */
+    auto modulesWithAbsPos() const -> std::ranges::range auto
```

Comment:
Nebylo by lepší přímo specifikovat šablonu konceptu než to nechávat na auto? Takto se musíš dívat do kódu, abys pochopil, co dostáváš.

Reply:
No, já nevím jak přesněji to specifikovat bez toho, abych zaváděl nový koncept...
dokumentace pro transform na cppreference je `constexpr ranges::view auto transform( R&& r, F&& fun );` ...
Nedošel jsem na to, že by šlo v konceptu specifikovat čeho to je range...

... Tak jsem se na to víc podíval a jediný `output_range` bere output typ, tak to předělám na něj - a případně to nepůjde (by se mi hodilo toho specifikovat mnohem víc ... třeba že je bidirectional + sized + output<...> - ale to bychom museli zavádět vlastní typy a nevím jak moc to dává smysl.

Reply:
EDIT: output_range asi neni the way to go - to není to, co chceme modelovat (my vracíme spíš input_range)

Přijde mi, že cpp ranges knihovna počítá s tím, že se budou používat ranges v argumentech a ne tolik pro return values

Reply:
můžeme si k tomu napsat vlastní tagy, ale je to víc práce, než co jsem chtěl v tomhle pull requestu

Reply:
Prostudoval jsem si ranges a máš pravdu. Nemá to elegantní řešení. Takže za mě to klidně takto nechejme.